test/regress_ws.c

```diff
@@ -340,17 +340,20 @@ http_ws_test(void *arg)
 {
 	struct basic_test_data *data = arg;
 	struct bufferevent *bev = NULL;
-	evutil_socket_t fd;
+	evutil_socket_t fd = EVUTIL_INVALID_SOCKET;
 	ev_uint16_t port = 0;
 	int ssl = 0;
 	struct evhttp *http = http_setup(&port, data->base, ssl);
+	if (!http) 
+		goto end;
 	struct evbuffer *out;
```

Comment:
It's C, not C++, so the order matters:
```suggestion
	struct evbuffer *out;
	if (!http) 
		goto end;
```